shell/common/shell.cc

```diff
@@ -1649,6 +1649,20 @@ double Shell::GetMainDisplayRefreshRate() {
   return display_manager_->GetMainDisplayRefreshRate();
 }
 
+void Shell::RegisterImageDecoder(ImageGeneratorFactory factory,
+                                 int32_t priority) {
+  FML_DCHECK(is_setup_);
```

Comment:
Shell API generally expects to be run on the platform task runner unless otherwise specified (we have a couple calls that need to run on the raster task runner for example).

Please add a DCHECK here that asserts that this is run on the platform task runner (see other methods in this file for examples).

Reply:
Done.